### PR TITLE
feat: add bacon and bacon-ls

### PR DESCRIPTION
**Pull request by crisidev**
## Describe your changes
Add support for [bacon](https://dystroy.org/bacon/) and [bacon-ls](https://github.com/crisidev/bacon-ls) ([blog post](https://lmno.lol/crisidev/bacon-language-server)).

It is already supported by [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/pull/3160).

## Checklist before requesting a review

- [x] I have successfully tested installation of the package.
- [x] I have successfully tested the package after installation.

## Screenshots
Screenshot 2024-05-17 at 20 11 35

### CODE REVIEW

File: packages/bacon/package.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+name: bacon
+description: Bacon is a background rust code checker
+homepage: https://github.com/Canop/bacon
+licenses:
+  - AGPL-3.0-or-later
+languages:
+  - Rust
+categories:
+  - Runtime
```
Comment (williamboman) [inline packages/bacon/package.yaml:10]: I think `Linter` would be a more appropriate category here? The `Runtime` category is not really used yet by Mason and is more geared towards software that provide a traditional runtime, like Node, Python, Deno, etc.

Comment (crisidev) [reply packages/bacon/package.yaml:10]: Sounds good, I was unsure about this as well. Let me fix it :)

Comment (crisidev) [reply packages/bacon/package.yaml]: Done in https://github.com/mason-org/mason-registry/pull/5774/commits/8838fbfe4b8406015979bb7c572e96b7dfe98128

**Review by williamboman — APPROVED** [review]
Thanks!

**Pull request merged**